### PR TITLE
Test generating itinerary in order and generating optimized itinerary

### PR DESCRIPTION
**Pull request by zhoua0805**
- Add two test cases to test for generating itinerary in order and generating optimized itinerary
- Use mockito to mock the function getDirectionRoute which uses the directions api
- TODO: test more edge cases such as no events, too many events, etc.

### CODE REVIEW

File: planet/src/main/java/com/google/planet/data/ItineraryGenerator.java [hunk 1]
```diff
@@ -84,7 +84,7 @@
     // Each DirectionsLeg contains starting address, ending address, and duration
     // DirectionRoute also has int[] waypointOrder, which will contain the order of the waypoints
     // For the MVP, real time traffic is NOT used.
-    private DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) throws ItineraryException{
+    public DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) throws ItineraryException{
```
Comment (azhang72) [inline planet/src/main/java/com/google/planet/data/ItineraryGenerator.java]: Instead of making a method public just so that its own test can mock it, we can have a helper class as a thin wrapper around Maps API. We pass the helper class into ItineraryGenerator in its constructor and mock the helper in tests.
The pattern is called [_dependency injection_](https://en.wikipedia.org/wiki/Dependency_injection) in case you want to read more about it.
I recommend you refactoring the code if you have time in your last week.

File: planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java [hunk 2]
```diff
@@ -20,8 +20,8 @@
 import com.google.planet.data.Event;
 import com.google.planet.data.ItineraryGenerator;
 import com.google.planet.data.ItineraryItem;
-import com.google.planet.data.ItineraryOrder;
 import com.google.planet.data.ItineraryException;
+import com.google.planet.data.ItineraryOrder;
```
Comment (azhang72) [inline planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java:24]: Do we need this?

File: planet/src/main/java/com/google/planet/data/ItineraryGenerator.java (hunk 1 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/data/ItineraryGenerator.java]: Ah okay, thank you!

File: planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java (hunk 2 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java:24]: ItineraryOrder has OPTIMIZED and UNOPTIMIZED which are boolean values to pass in the itinerary generator function. I used them instead of primitives (true/false) hoping to make the code more readable

Comment (azhang72) [reply planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java:24]: Why is the import line the only change in this file? Did you start using the enums in a previous change and it wouldn't build?

File: planet/src/main/java/com/google/planet/data/ItineraryGenerator.java (hunk 1 above)
Comment (azhang72) [reply planet/src/main/java/com/google/planet/data/ItineraryGenerator.java]: Also consider replace the boolean |optimized| with your enum in this file

File: planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java (hunk 2 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java:24]: Oh it was there before too hahah. I accidentally reordered it when resolving merge conflicts

File: planet/src/main/java/com/google/planet/data/ItineraryGenerator.java (hunk 1 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/data/ItineraryGenerator.java]: Fixed!

**Review by azhang72 — APPROVED** [review]

**Review by emilyzhaoo — APPROVED** [review]
LGTM
